onsset/onsset.py

```diff
@@ -358,14 +373,77 @@ def get_lcoe(self, energy_per_cell, people, num_people_per_hh, start_year, end_y
         investment_cost = pd.DataFrame(investment_cost)
         installed_capacity = pd.DataFrame(installed_capacity)
 
-        if get_max_dist:
-            return lcoe, investment_cost, installed_capacity, peak_load
-        elif self.hybrid:
-            hybrid_capacity = pd.DataFrame(self.hybrid_capacity)
-            return lcoe + pd.DataFrame(self.hybrid_fuel), pd.DataFrame(investment_cost[0] + self.hybrid_investment), \
-                hybrid_capacity
+        if genset_backup:
+            return  lcoe, investment_cost, installed_capacity, installed_capacity_diesel_genset, lcoe_diesel_genset
         else:
-            return lcoe, investment_cost, installed_capacity
+            if get_max_dist:
+                return lcoe, investment_cost, installed_capacity, peak_load
+            elif self.hybrid:
+                hybrid_capacity = pd.DataFrame(self.hybrid_capacity)
+                return lcoe + pd.DataFrame(self.hybrid_fuel), pd.DataFrame(investment_cost[0] + self.hybrid_investment), \
+                    hybrid_capacity
+            else:
+                return lcoe, investment_cost, installed_capacity
+
+    def get_lcoe_backup(self, project_life, step, unmet_demand, fuel_cost_settlement):
+
+
+        if type(unmet_demand) == int or type(unmet_demand) == float or type(unmet_demand) == np.float64:
+            if unmet_demand == 0:
+                # If there is no demand, set the demand low (prevent div/0 error) and continue.
+                unmet_demand = 0.000000000001
+        else:
+            unmet_demand = np.maximum(unmet_demand, 0.000000000001)
+
+        reinvest_year = 0
+
+        # If the technology life is less than the project life, we will have to invest twice to buy it again
+        if self.tech_life + step < project_life:
+            reinvest_year = self.tech_life + step
+
+        year = np.arange(project_life)
+        discount_factor = (1 + self.discount_rate) ** year
+
+        installed_capacity_diesel_genset = unmet_demand / self.capacity_factor / HOURS_PER_YEAR / self.base_to_peak_load_ratio
```

Comment:
I think the diesel generator capacity should be sized based on the total demand in the settlement (i.e. if reliability is 90% we size it to cover the demand for 10% of the hours, rather than 10% of the load all of the time). So either use pass the full demand into the function as well, or adjust the hours in the denominator.